### ISSUE TITLE
Glitch effect: app window zoom bug

### ISSUE THREAD

**Issue by Erwin-Iosef** [Schneegans/Burn-My-Windows#481]
## Description
Whenever I open KDE apps like `Ark` or `systemsettings` the effect zooms into the contents of the window too much before resetting into a normal size. Like this.

https://github.com/user-attachments/assets/258c49d0-31a5-4e92-990c-848ac83a7734

A minor bug to be sure.

1. Install glitch effect from kde store
2. Open system settings
4. See bug

## System
_Please complete the following information:_
 - Linux distribution : Arch Linux
 - Burn-My-Windows version 44 for Kwin 6 from the kde store
- KDE Plasma 6.2.4

**Comment by Schneegans** [comment]
Thanks for the report! Is this limited to a few applications or does it happen with all of them? Also, is it limited to the glitch effect or does it happen with other Burn-My-Windows effects as well?

**Comment by Erwin-Iosef** [comment]
Well Ark glitches out as well and I think no. It only happens either with `systemsettings` or its applets like `KDE wallet`.

**Comment by Erwin-Iosef** [comment]
It only happens when it's fullscreen I believe. Unfortunately I don't know if I can open them windowed. I tested fullscreen apps like OBS and Firefox, they work but firefox doesn't open(at least a new window) with the effect but does close with it.

**Comment by Erwin-Iosef** [comment]
Hello, just popping in to report that this bug also happens in other Burn-my-windows effects such as TV Glitch and Incinerate.

**Comment by Erwin-Iosef** [comment]
@Schneegans I've been slowly looking through the code section `code/main.js`, you think we should maybe blacklist the plasma apps or something? This issue seems to only happen with plasma apps right now.

**Comment by Erwin-Iosef** [comment]
@Schneegans Just a thought, but is there a way for desktop effects to control whether certain windows maximizes or restores on opening them, I seem to have a vague memory or thought that making the plasma apps(systemsettings) issue be fixed by having them be opened at windowed size.

**Comment by Schneegans** [comment]
Well, the second video you have shown is definitely a limitation of KWin and how shader effects work there. The problem is that a shader effect is only applied to the first frame an application renders. So a window pops up and the initial content is used during the entire animation. Only when the animation is done, the window content will be updated.

If a window opens at a different size but then instantly resizes itself, I guess also your first video can be explained.

I once talked to the KWin developers on matrix and they said that this is indeed a limitation which will be very hard to solve.

I think [this issue](https://bugs.kde.org/show_bug.cgi?id=463845) has the same underlying issue. But maybe you could open a KWin bug report which sheds some additional light on the issue as it's not only a problem for the window decorations but for many applications. I think I also observed this in the file browser where the files of a directory only become visible after the animation is done.

**Comment by Erwin-Iosef** [comment]
> Well, the second video you have shown is definitely a limitation of KWin and how shader effects work there. The problem is that a shader effect is only applied to the first frame an application renders. So a window pops up and the initial content is used during the entire animation. Only when the animation is done, the window content will be updated.

I agree, I recorded opening ark with no effects and played it back slowly and the glitch was there, it was just too fast to be barely noticeable.

> If a window opens at a different size but then instantly resizes itself, I guess also your first video can be explained.
I once talked to the KWin developers on matrix and they said that this is indeed a limitation which will be very hard to solve.

I see, I knew it in my gut it was a Plasma problem. The question is how hard it'll be for a fix since the source of the issue is more upstream.

> I think [this issue](https://bugs.kde.org/show_bug.cgi?id=463845) has the same underlying issue. But maybe you could open a KWin bug report which sheds some additional light on the issue as it's not only a problem for the window decorations but for many applications. I think I also observed this in the file browser where the files of a directory only become visible after the animation is done.

I will try and I don't think that bug report is going to be seen since it's old version now. Thank you for giving me this information. Have a nice day :)

**Comment by Erwin-Iosef** [comment]
Bug reported at https://bugs.kde.org/show_bug.cgi?id=498557, would appreciate it if you can provide any extra info or insight.

**Comment by Gapoly** [comment]
I same the problem as you. Also when I open and close a windows, you can the windows tray before the animation finishes

https://github.com/user-attachments/assets/6fed23d2-2147-4403-9029-bb9f9e2dd75c

**Comment by Erwin-Iosef** [comment]
Yeah it's really annoying, read the above comments, it's a kwin bug actually. I reported it in the link above, maybe you can leave a comment?

**Comment by Gapoly** [comment]
@Erwin-Iosef I've added a comment from my experience as well. I really hope they fix the bug because I'd like to use the effects from Burn my windows.

**Comment by Erwin-Iosef** [comment]
Me too, thanks @Gapoly and we've done the most we could, it's already been confirmed. Now the rest is upto developers, it's probably a hard bug to fix as well.

**Comment by Erwin-Iosef** [comment]
Update for the reader: Change rendering backend to vulkan in Plasma Renderer(find it in application launcher/kickoff menu) for a fix.
Ref: https://bugs.kde.org/show_bug.cgi?id=499101
